src/core/net/ip4_headers.hpp

```diff
@@ -0,0 +1,470 @@
+     */
+    void SetVersionIHL(uint8_t aVersionIHL) { mVersIHL = aVersionIHL; }
+
+    /**
+     * This method indicates whether or not the IPv4 Version is set to 6.
+     *
+     * @retval TRUE   If the IPv4 Version is set to 4.
+     * @retval FALSE  If the IPv4 Version is not set to 4.
+     *
+     */
+    bool IsVersion4(void) const { return (mVersIHL & kVersionMask) == kVersion4; }
+
+    /**
+     * This method returns the octet for DSCP + ECN.
+     *
+     * @retval The octet for DSCP and ECN.
+     *
+     */
+    uint8_t GetDscpEcn() const { return mDscpEcn; }
+
+    /**
+     * This method gets the 6-bit Differentiated Services Code Point (DSCP) from Traffic Class field.
+     *
+     * @returns The DSCP value.
+     *
+     */
+    uint8_t GetDscp(void) const { return (mDscpEcn & kDscpMask) >> kDscpOffset; }
+
+    /**
+     * This method sets 6-bit Differentiated Services Code Point (DSCP) in IPv4 header.
+     *
+     * @param[in]  aDscp  The DSCP value.
+     *
+     */
+    void SetDscp(uint8_t aDscp) { mDscpEcn = static_cast<uint8_t>((mDscpEcn & ~kDscpMask) | (aDscp << kDscpOffset)); }
+
+    /**
+     * This method gets the 2-bit Explicit Congestion Notification (ECN) from Traffic Class field.
+     *
+     * @returns The ECN value.
+     *
+     */
+    Ecn GetEcn(void) const { return static_cast<Ecn>(mDscpEcn & kEcnMask); }
+
+    /**
+     * This method sets the 2-bit Explicit Congestion Notification (ECN) in IPv4 header..
+     *
+     * @param[in]  aEcn  The ECN value.
+     *
+     */
+    void SetEcn(Ecn aEcn) { mDscpEcn = ((mDscpEcn & ~kEcnMask) | aEcn); }
+
+    /**
+     * This method returns the IPv4 Payload Length value.
+     *
+     * @returns The IPv4 Payload Length value.
+     *
+     */
+    uint16_t GetTotalLength(void) const { return HostSwap16(mTotalLength); }
+
+    /**
+     * This method sets the IPv4 Payload Length value.
+     *
+     * @param[in]  aLength  The IPv4 Payload Length value.
+     *
+     */
+    void SetTotalLength(uint16_t aLength) { mTotalLength = HostSwap16(aLength); }
+
+    /**
+     * This method returns the IPv4 payload protocol.
+     *
+     * @returns The IPv4 payload protocol value.
+     *
+     */
+    uint8_t GetProtocol(void) const { return mProtocol; }
+
+    /**
+     * This method sets the IPv4 payload protocol.
+     *
+     * @param[in]  aNextHeader  The IPv4 payload protocol.
+     *
+     */
+    void SetProtocol(uint8_t aProtocol) { mProtocol = aProtocol; }
+
+    /**
+     * This method sets the IPv4 header checksum, the checksum is in network endian.
+     *
+     * @param[in] aChecksum The checksum for the IPv4 header.
+     */
+    void SetChecksum(uint16_t aChecksum) { mHeaderChecksum = aChecksum; }
+
+    /**
+     * This method returns the IPv4 Identification value.
+     *
+     * @returns The IPv4 Identification value.
+     */
+    uint16_t GetIdentification() { return HostSwap16(mIdentification); }
```

Comment:
add `(void)` and defined as `const`.
```suggestion
     *
     */
    uint16_t GetIdentification(void) const { return HostSwap16(mIdentification); }
```